src/live_restore/live_restore_server.c

```diff
@@ -0,0 +1,293 @@
+    ret = __wti_live_restore_fs_fill_holes(fh, wt_session);
+    WT_TRET(cursor->close(cursor));
+
+    return (ret);
+}
+
+/*
+ * __live_restore_populate_queue --
+ *     Populate the live restore work queue. Free all objects on failure.
+ */
+static int
+__live_restore_populate_queue(WT_SESSION_IMPL *session, uint64_t *work_count)
+{
+    WT_DECL_RET;
+    WT_LIVE_RESTORE_SERVER *server = &S2C(session)->live_restore_server;
+
+    TAILQ_INIT(&server->work_queue);
+    /*
+     * Open a metadata cursor to gather the list of objects. The metadata file is built from the
+     * WiredTiger.backup file, during __wt_turtle_init. Thus this function must be run after that
+     * function. I don't know if we have a way of enforcing that.
+     */
+
+    /*
+     * FIXME-WT-13888: Add logic to queue log files first, then the oplog then the history store.
+     * This will use a directory list call.
+     */
+    WT_CURSOR *cursor;
+    WT_RET(__wt_metadata_cursor(session, &cursor));
+    WT_LIVE_RESTORE_WORK_ITEM *work_item = NULL;
+    __wt_verbose_debug1(session, WT_VERB_FILEOPS, "Initializing the live restore work %s", "queue");
+
+    *work_count = 0;
+    while ((ret = cursor->next(cursor)) == 0) {
+        const char *uri = NULL;
+        WT_ERR(cursor->get_key(cursor, &uri));
+        if (WT_PREFIX_MATCH(uri, "file:")) {
+            __wt_verbose_debug2(
+              session, WT_VERB_FILEOPS, "Adding an item to the work queue %s", uri);
+            WT_ERR(__wt_calloc_one(session, &work_item));
+            WT_ERR(__wt_strdup(session, uri, &work_item->uri));
+            TAILQ_INSERT_HEAD(&server->work_queue, work_item, q);
+            (*work_count)++;
+        }
+    }
+    WT_ERR_NOTFOUND_OK(ret, false);
+
+    if (0) {
+err:
+        /* The queue insert cannot fail so whatever we've put into it is guaranteed to be there. */
+        __live_restore_work_queue_drain(session);
+
+        /* Free the various items. */
+        __wt_free(session, work_item->uri);
+        __wt_free(session, work_item);
+    }
+    return (ret);
+}
+
+/*
+ * __wt_live_restore_server_create --
+ *     Start the worker threads and build the work queue.
+ */
+int
+__wt_live_restore_server_create(WT_SESSION_IMPL *session, const char *cfg[])
+{
+    WT_CONFIG_ITEM cval;
+
+    WT_LIVE_RESTORE_SERVER *server = &S2C(session)->live_restore_server;
+
+    /* Check that we have a live restore file system before starting the threads. */
+    if (!F_ISSET(S2C(session), WT_CONN_LIVE_RESTORE_FS))
+        return (0);
+
+    /*
+     * Set this state before we run the threads, if we do it after there's a chance we'll context
+     * switch and then this state will happen after the finish state. This also means we transition
+     * through all valid states.
+     */
+    WT_STAT_CONN_SET(session, live_restore_state, WT_LIVE_RESTORE_IN_PROGRESS);
+
+    /* Read the threads_max config, 0 threads is valid in which case we don't do anything. */
```

Comment:
What's the use case for 0 background threads, is it temporary until we've implemented concurrency in WT-13825?

Reply:
Being able to specify zero background threads is useful for that reason, yes. But also Sue described a use case where backup may want to "warm up" a host to speed up the restore process. In that case they aren't interested in transferring all the files over but instead want to pre-run WT recovery and oplog replay.

Reply:
0 background threads is also useful for testing, with one config we turn off half our functionality making it a lot more deterministic.